packages/labs/motion/src/spring.ts

```diff
@@ -0,0 +1,175 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+import {
+  ReactiveController,
+  ReactiveControllerHost,
+} from '@lit/reactive-element';
+import {Spring, SpringConfig as WobbleSpringConfig} from 'wobble';
+
+export type SpringConfig = Partial<WobbleSpringConfig>;
+
+/**
+ * A reactive controller that implements a 1D spring physics simulation based
+ * on the equations behind damped harmonic oscillators
+ * (https://en.wikipedia.org/wiki/Harmonic_oscillator#Damped_harmonic_oscillator).
+ */
+export class SpringController implements ReactiveController {
+  private _host: ReactiveControllerHost & HTMLElement;
+
+  protected _spring: Spring;
+  private _toValue: number;
+
+  constructor(
+    host: ReactiveControllerHost & HTMLElement,
+    options?: SpringConfig
+  ) {
+    (this._host = host).addController(this);
+    this._toValue = options?.toValue ?? 0;
+    this._spring = new Spring(options);
+    this._spring.onUpdate(() => this._host.requestUpdate());
+  }
+
+  get currentValue() {
+    return this._spring.currentValue;
+  }
+
+  /**
+   * The spring's current velocity in units / ms.
+   */
+  get currentVelocity(): number {
+    return this._spring.currentVelocity;
+  }
+
+  /**
+   * If the spring has reached its `toValue`, or if its velocity is below the
+   * `restVelocityThreshold`, it is considered at rest. If `stop()` is called
+   * during a simulation, both `isAnimating` and `isAtRest` will be false.
+   */
+  get isAtRest(): boolean {
+    return this._spring.isAtRest;
+  }
+
+  /**
+   * Whether or not the spring is currently emitting values.
+   *
+   * Note: this is distinct from whether or not it is at rest.
+   * See also `isAtRest`.
+   */
+  get isAnimating(): boolean {
+    return this._spring.isAnimating;
+  }
+
+  get toValue() {
+    return this._toValue;
+  }
+
+  set toValue(toValue: number) {
+    this._toValue = toValue;
+    this._spring.updateConfig({toValue});
+    if ((this._host as HTMLElement).isConnected) {
```

Comment:
Does this need the type assertion since it has type: `ReactiveControllerHost & HTMLElement`?

Reply:
fixed